PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:13 + add:0, change matches:
-    _inherit = 'account.move'
-    _inherit = "account.move.line"
-    _inherit = 'account.analytic.account'
-    _inherit = 'account.analytic.line'
-    _inherit = 'account.analytic.applicability'
-    _inherit = 'mrp.production'
-    _inherit = 'mrp.routing.workcenter'
-    _inherit = 'mrp.workcenter.productivity'
-    _inherit = 'mrp.workorder'
-    _inherit = 'product.template'
-    _inherit = 'product.product'
-    _inherit = 'product.category'
-    _inherit = "stock.move"

Total Changes: 31

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/mrp_account/models/account_move.py
+++ b/addons/mrp_account/models/account_move.py
@@ -7,7 +7,7 @@ from collections import defaultdict
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     wip_production_ids = fields.Many2many(
         'mrp.production', string="Relevant WIP MOs",
@@ -40,7 +40,7 @@ class AccountMove(models.Model):
 
 
 class AccountMoveLine(models.Model):
-    _inherit = "account.move.line"
+    _inherit = ["account.move.line"]
 
     def _get_invoiced_qty_per_product(self):
         # Replace the kit-type products with their components

--- a/addons/mrp_account/models/analytic_account.py
+++ b/addons/mrp_account/models/analytic_account.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 
 
 class AccountAnalyticAccount(models.Model):
-    _inherit = 'account.analytic.account'
+    _inherit = ['account.analytic.account']
     _description = 'Analytic Account'
 
     production_ids = fields.Many2many('mrp.production')
@@ -74,12 +74,13 @@ class AccountAnalyticAccount(models.Model):
 
 
 class AccountAnalyticLine(models.Model):
-    _inherit = 'account.analytic.line'
+    _inherit = ['account.analytic.line']
 
     category = fields.Selection(selection_add=[('manufacturing_order', 'Manufacturing Order')])
 
+
 class AccountAnalyticApplicability(models.Model):
-    _inherit = 'account.analytic.applicability'
+    _inherit = ['account.analytic.applicability']
     _description = "Analytic Plan's Applicabilities"
 
     business_domain = fields.Selection(

--- a/addons/mrp_account/models/mrp_production.py
+++ b/addons/mrp_account/models/mrp_production.py
@@ -8,7 +8,7 @@ from odoo.tools import float_round
 
 
 class MrpProduction(models.Model):
-    _inherit = 'mrp.production'
+    _inherit = ['mrp.production']
 
     extra_cost = fields.Float(copy=False, string='Extra Unit Cost')
     show_valuation = fields.Boolean(compute='_compute_show_valuation')

--- a/addons/mrp_account/models/mrp_routing.py
+++ b/addons/mrp_account/models/mrp_routing.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class MrpRoutingWorkcenter(models.Model):
-    _inherit = 'mrp.routing.workcenter'
+    _inherit = ['mrp.routing.workcenter']
 
     def _total_cost_per_hour(self):
         self.ensure_one()

--- a/addons/mrp_account/models/mrp_workcenter.py
+++ b/addons/mrp_account/models/mrp_workcenter.py
@@ -5,7 +5,6 @@ from odoo import api, models, fields
 
 
 class MrpWorkcenter(models.Model):
-    _name = 'mrp.workcenter'
     _inherit = ['mrp.workcenter', 'analytic.mixin']
 
     costs_hour_account_ids = fields.Many2many('account.analytic.account', compute="_compute_costs_hour_account_ids", store=True)
@@ -21,7 +20,6 @@ class MrpWorkcenter(models.Model):
 
 
 class MrpWorkcenterProductivity(models.Model):
-    _name = 'mrp.workcenter.productivity'
-    _inherit = 'mrp.workcenter.productivity'
+    _inherit = ['mrp.workcenter.productivity']
 
     account_move_line_id = fields.Many2one('account.move.line')

--- a/addons/mrp_account/models/mrp_workorder.py
+++ b/addons/mrp_account/models/mrp_workorder.py
@@ -5,7 +5,7 @@ from odoo import fields, models, _
 
 
 class MrpWorkorder(models.Model):
-    _inherit = 'mrp.workorder'
+    _inherit = ['mrp.workorder']
 
     mo_analytic_account_line_ids = fields.Many2many('account.analytic.line', 'mrp_workorder_mo_analytic_rel', copy=False)
     wc_analytic_account_line_ids = fields.Many2many('account.analytic.line', 'mrp_workorder_wc_analytic_rel', copy=False)

--- a/addons/mrp_account/models/product.py
+++ b/addons/mrp_account/models/product.py
@@ -6,8 +6,7 @@ from odoo.tools import float_round, groupby
 
 
 class ProductTemplate(models.Model):
-    _name = 'product.template'
-    _inherit = 'product.template'
+    _inherit = ['product.template']
 
     def _get_product_accounts(self):
         accounts = super()._get_product_accounts()
@@ -28,8 +27,7 @@ class ProductTemplate(models.Model):
 
 
 class ProductProduct(models.Model):
-    _name = 'product.product'
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     def button_bom_cost(self):
         self.ensure_one()
@@ -114,7 +112,7 @@ class ProductProduct(models.Model):
 
 
 class ProductCategory(models.Model):
-    _inherit = 'product.category'
+    _inherit = ['product.category']
 
     property_stock_account_production_cost_id = fields.Many2one(
         'account.account', 'Production Account', company_dependent=True, ondelete='restrict',

--- a/addons/mrp_account/models/stock_move.py
+++ b/addons/mrp_account/models/stock_move.py
@@ -7,7 +7,7 @@ from odoo.tools import float_round
 
 
 class StockMove(models.Model):
-    _inherit = "stock.move"
+    _inherit = ["stock.move"]
 
     def _filter_anglo_saxon_moves(self, product):
         res = super(StockMove, self)._filter_anglo_saxon_moves(product)
